### ISSUE TITLE
[Wechat]unhandled promise rejection<TypeError: undefined is not an object(evalating 'charJson.strokes.map')>

### ISSUE THREAD

**Issue by bravekingzhang** [chanind/hanzi-writer#121]
**The simulator is OK,** 

![image](https://user-images.githubusercontent.com/4476322/59507732-7ea2e780-8ede-11e9-8eef-a3f8f557b129.png)

**But on my ios phone is Not very well**
![image](https://user-images.githubusercontent.com/4476322/59507562-ed337580-8edd-11e9-89c5-39f4c105d4bf.png)

**Comment by bravekingzhang** [comment]
**on my android phone is ok**

**Comment by chanind** [comment]
How can I reproduce this error? It needs to be installed on iPhone Wechat?

**Comment by bravekingzhang** [comment]
yes ,you should preview `wechat mini programe` on iphone wechat

**Comment by bravekingzhang** [comment]
![image](https://user-images.githubusercontent.com/4476322/59545467-2527d080-8f51-11e9-822d-85f152b67b6d.png)
both are ok

**Comment by chanind** [comment]
I don't have an iPhone so I can't test this unfortunately. Could you try adding setting `charDataLoader` to the following in `createHanziWriterContext`? That will log to the console the data that gets returned:

```
createHanziWriterContext({
  id: 'your-id',
  page: this,
  // other params here
  charDataLoader:   (char, onLoad, onError) => {
    const VERSION = '2.0';
    const url = `https://cdn.jsdelivr.net/npm/hanzi-writer-data@${VERSION}/${char}.json`;
    wx.request({
      url,
      header: {
        'content-type': 'application/json',
      },
      success: (res) => {
        console.log(res);
        onLoad(res.data);
      },
      fail: onError,
    });
  }
});
```

**Comment by bravekingzhang** [comment]
sorry for so late to reply you @chanind 

**your website has been added to my config**

![image](https://user-images.githubusercontent.com/4476322/59574742-00676080-90eb-11e9-8e88-1316637fcce0.png)
 and this is the **console.log**
![image](https://user-images.githubusercontent.com/4476322/59574655-b2525d00-90ea-11e9-8280-2272629f5c89.png)

**Comment by bravekingzhang** [comment]
I found this https://developers.weixin.qq.com/community/develop/doc/000446aaeb83b87eea28a65035b400?_at=1558851206244

and I try encodeURI for "hanzi"
```js
this.writerCtx = createHanziWriterContext({
					id: 'hz-writer',
					character: encodeURI(this.character),
					page: this,
					showHintAfterMisses: 1,
				});
```
the request will ok , I think ios not support for url contains chinese character.

but a new problem coming...

wechat developer tool run like this :
![image](https://user-images.githubusercontent.com/4476322/59575818-92716800-90ef-11e9-8096-afffd59d5441.png)
but ios phone run like this:
![image](https://user-images.githubusercontent.com/4476322/59575871-d3697c80-90ef-11e9-87b5-257413804860.png)
does the canvas api not fit ios phone?

**Comment by chanind** [comment]
I just released version `1.0.3` of the miniprogram plugin, which fixes the `encodeURI` issue you found - thanks for finding that! For the canvas rendering too large, I'm not sure what the issue could be. You can control how large it renders by setting `width` and `height` in the `<hanzi-writer-view>`, so you can try setting a smaller number for width and height if the screen is too small?

**Comment by bravekingzhang** [comment]
![image](https://user-images.githubusercontent.com/4476322/59990868-80577280-9677-11e9-8576-242d2617cf90.png)
one is 100 * 100 ,and the anther 300 * 300
but they both  have the same question

**Comment by chanind** [comment]
Which version of iPhone / iOS is it? I'll try to get a test device

hz <notifications@github.com> 于 2019年6月24日周一 12:00写道：

**Comment by bravekingzhang** [comment]
iOS 12.3.1
wechat 7.0.4

**Comment by kevin82008** [comment]
> I just released version `1.0.3` of the miniprogram plugin, which fixes the `encodeURI` issue you found - thanks for finding that! For the canvas rendering too large, I'm not sure what the issue could be. You can control how large it renders by setting `width` and `height` in the `<hanzi-writer-view>`, so you can try setting a smaller number for width and height if the screen is too small?


The reason may be：
@HanziWriterRenderer.js,  In the following statement(in method _animationFrame)，the value of Y axis   is less than zero, the Performance on ios is wrong: 
ctx.scale(this._positioner.scale, -1 * this._positioner.scale);

**Comment by kevin82008** [comment]
> I just released version `1.0.3` of the miniprogram plugin, which fixes the `encodeURI` issue you found - thanks for finding that! For the canvas rendering too large, I'm not sure what the issue could be. You can control how large it renders by setting `width` and `height` in the `<hanzi-writer-view>`, so you can try setting a smaller number for width and height if the screen is too small?

in HanziWriterRenderer.js, the following statement(in method _animationFrame)：
ctx.scale(this._positioner.scale, -1 * this._positioner.scale);

should be replaced by:
ctx.transform(1, 0, 0, -1, 0, 0);
ctx.scale(this._positioner.scale, this._positioner.scale);

it works well in IOS
